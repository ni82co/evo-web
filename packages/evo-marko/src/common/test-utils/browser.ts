import { fireEvent, createEvent } from "@marko/testing-library";
/**
 * Simulates a touch based scroll event over 4 animation frames.
 *
 * @param {HTMLElement} el The element to scroll.
 * @param {number} to The new scrollLeft for the element.
 * @param {function} cb A callback to call after the scroll.
 */
function simulateScroll(el: HTMLElement, to: number, cb: Function) {
  fireEvent.scroll(el);
  el.scrollLeft = to;
  setTimeout(cb, 600);
}

function waitFrames(count: number, cb: Function) {
  if (count) {
    return requestAnimationFrame(() => {
      waitFrames(count - 1, cb);
    });
  }

  cb();
}

interface FastAnimations {
  fastAnimationStyle?: HTMLStyleElement;
  start: () => void;
  stop: () => void;
}

const fastAnimations:FastAnimations = {
  // Adds an style to the document which forces all transitions to run more quickly for the tests.
  start() {
    if (this.fastAnimationStyle) {
      return;
    }

    this.fastAnimationStyle = document.createElement("style");
    this.fastAnimationStyle.innerHTML = `* {
                transition-duration: 0.1s !important;
                transition-delay: 0.1s !important;
                scroll-behavior: auto !important;
            }`;
    document.head.appendChild(this.fastAnimationStyle);
  },
  stop() {
    if (!this.fastAnimationStyle) {
      return;
    }

    document.head.removeChild(this.fastAnimationStyle);
    this.fastAnimationStyle = undefined;
  },
};

export { fastAnimations, waitFrames, simulateScroll };
