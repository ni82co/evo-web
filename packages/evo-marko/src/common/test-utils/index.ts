export * from "./browser"
export * from "./snapshots";