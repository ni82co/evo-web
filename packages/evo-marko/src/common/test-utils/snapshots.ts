import { render, normalize } from "@marko/testing-library";
import { expect } from "vitest";

async function snapshotHTML(template: any, input?: any) {
  const { container } = await render(template, input);
  expect(normalize(container)).toMatchSnapshot();
}

export { snapshotHTML };
