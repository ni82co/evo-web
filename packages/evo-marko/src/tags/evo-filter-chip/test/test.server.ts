import { describe, it } from "vitest";

import { composeStories } from "@storybook/marko";
import { snapshotHTML } from "../../../common/test-utils/snapshots";
import * as stories from "../filter-chip.stories"; // import all stories from the stories file
const { Default, Expressive, MenuButton } = composeStories(stories);

describe("evo-filter-chip", () => {
  it("renders static default", async () => {
    await snapshotHTML(Default);
  });

  it("renders expressive", async () => {
    await snapshotHTML(Expressive);
  });

  it("renders menu button", async () => {
    await snapshotHTML(MenuButton);
  });

  it("renders static selected", async () => {
    await snapshotHTML(Default, { selected: true });
  });

  it("renders expressive selected", async () => {
    await snapshotHTML(Expressive, { selected: true });
  });

  it("renders menu button selected", async () => {
    await snapshotHTML(MenuButton, { selected: true });
  });

  it("renders menu button expanded", async () => {
    await snapshotHTML(MenuButton, { expanded: true });
  });

  it("renders menu button expanded and selected", async () => {
    await snapshotHTML(MenuButton, { expanded: true, selected: true });
  });

  it("renders as link", async () => {
    await snapshotHTML(Default, { href: "www.ebay.com" });
  });

  it("renders as link selected", async () => {
    await snapshotHTML(Expressive, {
      href: "www.ebay.com",
      a11ySelectedText: "selected",
      selected: true,
    });
  });
});
