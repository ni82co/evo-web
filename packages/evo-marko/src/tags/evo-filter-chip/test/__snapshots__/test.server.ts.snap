// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`evo-filter-chip > renders as link 1`] = `
<DocumentFragment>
  <button
    aria-pressed="false"
    class="filter-chip"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="icon icon--16"
    >
      <symbol
        id="GENERATED-0"
        viewBox="0 0 16 16"
      >
        <path
          clip-rule="evenodd"
          d="M4.17 3A3.001 3.001 0 0 1 7 1c1.826 0 2.667 1.549 2.97 2.757l.394 1.575 4.073 2.222A3 3 0 0 1 16 10.187V11a3 3 0 0 1-3 3H2.5A2.5 2.5 0 0 1 0 11.5v-6A2.5 2.5 0 0 1 2.5 3h1.67ZM6 4a1 1 0 0 1 1-1c.629 0 .903.737 1.03 1.243l.263 1.05-1 1a1 1 0 0 0 1.414 1.414l.648-.648c.248.145.505.277.758.414l-.82.82a1 1 0 0 0 1.414 1.414l1.236-1.236 1.536.838a1 1 0 0 1 .521.878V11a1 1 0 0 1-1 1H7v-.5A2.5 2.5 0 0 0 4.5 9H2V8h1.5A2.5 2.5 0 0 0 6 5.5V4Zm-1 8v-.5a.5.5 0 0 0-.5-.5H2v.5a.5.5 0 0 0 .5.5H5ZM2.5 5H4v.5a.5.5 0 0 1-.5.5H2v-.5a.5.5 0 0 1 .5-.5Z"
          fill-rule="evenodd"
        />
      </symbol>
      <use
        href="#icon-sneaker-16"
      />
    </svg>
    <span
      class="filter-chip__text"
    >
      Filter
    </span>
  </button>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.e=[0,1,_.a={"EventAttributes:#text/0":_.b={},"ConditionalScope:#text/0":_.d={},"ConditionalRenderer:#text/0":"button",content:_._["src/tags/evo-filter-chip/examples/default.marko_1_renderer"](_.c={}),icon:{content:_._["src/tags/evo-filter-chip/examples/default.marko_2_renderer"](_.c),*[Symbol.iterator](){yield this}},variant:"default",htmlInput:{},isMenu:!1,mounted:!1,"ClosureScopes:selected":_.j=new Set},_.d,_.g={_:_.a,"ClosureSignalIndex:selected":0,"#ClosestBranchId":3},3,{htmlInput:{}}],_.b.click=_._["src/tags/evo-filter-chip/index.marko_0/onClick"](_.a),(_.f=new Set).add(_.g),(_.h=new Set).add(_.g),(_.i=new Set).add(_.g),(_.j).add(_.g),_.e),"src/tags/evo-icon/index.marko_0_htmlInput",8,"src/tags/evo-filter-chip/index.marko_0",2];M._.w()
  </script>
</DocumentFragment>
`;

exports[`evo-filter-chip > renders as link selected 1`] = `
<DocumentFragment>
  <a
    class="filter-chip filter-chip--expressive filter-chip--selected"
  >
    <span
      class="filter-chip__media"
    >
      <img
        alt="image"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile2.png"
      />
    </span>
    <span
      class="filter-chip__text"
    >
      Filter
      <span
        class="clipped"
      >
         - selected
      </span>
    </span>
  </a>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.g=[0,1,_.a={"EventAttributes:#text/0":_.b={},"ConditionalScope:#text/0":_.d={},"ConditionalRenderer:#text/0":"a",content:_._["src/tags/evo-filter-chip/examples/expressive.marko_1_renderer"](_.c={}),image:{alt:"image",src:"https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile2.png",*[Symbol.iterator](){yield this}},a11ySelectedText:"selected",variant:"expressive",htmlInput:{},isMenu:!1,isAnchor:!0,mounted:!1,selected:!0,"ClosureScopes:selected":_.n=new Set},_.d,_.e={"ConditionalRenderer:#text/2":0,"ConditionalScope:#text/2":_.f={},_:_.a,"ClosureSignalIndex:selected":0,"#ClosestBranchId":3},_.i={_:_.e},1,_.f],_.b.click=_._["src/tags/evo-filter-chip/index.marko_0/onClick"](_.a),(_.h=new Set).add(_.i),(_.j=new Set).add(_.f),(_.k=new Set).add(_.e),(_.l=new Set).add(_.e),(_.m=new Set).add(_.e),(_.n).add(_.e),_.g),"src/tags/evo-filter-chip/index.marko_3_image",5,"src/tags/evo-filter-chip/index.marko_0",2];M._.w()
  </script>
</DocumentFragment>
`;

exports[`evo-filter-chip > renders expressive 1`] = `
<DocumentFragment>
  <button
    aria-pressed="false"
    class="filter-chip filter-chip--expressive"
    type="button"
  >
    <span
      class="filter-chip__media"
    >
      <img
        alt="image"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile2.png"
      />
    </span>
    <span
      class="filter-chip__text"
    >
      Filter
    </span>
  </button>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.f=[0,1,_.a={"EventAttributes:#text/0":_.b={},"ConditionalScope:#text/0":_.d={},"ConditionalRenderer:#text/0":"button",content:_._["src/tags/evo-filter-chip/examples/expressive.marko_1_renderer"](_.c={}),image:{alt:"image",src:"https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile2.png",*[Symbol.iterator](){yield this}},variant:"expressive",htmlInput:{},isMenu:!1,isAnchor:!1,mounted:!1,"ClosureScopes:selected":_.l=new Set},_.d,_.e={_:_.a,"ClosureSignalIndex:selected":0,"#ClosestBranchId":3},_.h={_:_.e}],_.b.click=_._["src/tags/evo-filter-chip/index.marko_0/onClick"](_.a),(_.g=new Set).add(_.h),(_.i=new Set).add(_.e),(_.j=new Set).add(_.e),(_.k=new Set).add(_.e),(_.l).add(_.e),_.f),"src/tags/evo-filter-chip/index.marko_3_image",5,"src/tags/evo-filter-chip/index.marko_0",2];M._.w()
  </script>
</DocumentFragment>
`;

exports[`evo-filter-chip > renders expressive selected 1`] = `
<DocumentFragment>
  <button
    aria-pressed="true"
    class="filter-chip filter-chip--expressive"
    type="button"
  >
    <span
      class="filter-chip__media"
    >
      <img
        alt="image"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile2.png"
      />
    </span>
    <span
      class="filter-chip__text"
    >
      Filter
    </span>
  </button>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.f=[0,1,_.a={"EventAttributes:#text/0":_.b={},"ConditionalScope:#text/0":_.d={},"ConditionalRenderer:#text/0":"button",content:_._["src/tags/evo-filter-chip/examples/expressive.marko_1_renderer"](_.c={}),image:{alt:"image",src:"https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile2.png",*[Symbol.iterator](){yield this}},variant:"expressive",htmlInput:{},isMenu:!1,isAnchor:!1,mounted:!1,selected:!0,"ClosureScopes:selected":_.l=new Set},_.d,_.e={_:_.a,"ClosureSignalIndex:selected":0,"#ClosestBranchId":3},_.h={_:_.e}],_.b.click=_._["src/tags/evo-filter-chip/index.marko_0/onClick"](_.a),(_.g=new Set).add(_.h),(_.i=new Set).add(_.e),(_.j=new Set).add(_.e),(_.k=new Set).add(_.e),(_.l).add(_.e),_.f),"src/tags/evo-filter-chip/index.marko_3_image",5,"src/tags/evo-filter-chip/index.marko_0",2];M._.w()
  </script>
</DocumentFragment>
`;

exports[`evo-filter-chip > renders menu button 1`] = `
<DocumentFragment>
  <button
    aria-expanded="false"
    aria-pressed="false"
    class="filter-chip"
    type="button"
  >
    <span
      class="filter-chip__text"
    >
      Filter
    </span>
    <svg
      aria-hidden="true"
      class="filter-chip__trailing icon icon--12"
    >
      <symbol
        id="GENERATED-0"
        viewBox="0 0 12 12"
      >
        <path
          clip-rule="evenodd"
          d="M1.808 4.188a.625.625 0 0 1 .884 0L6 7.495l3.308-3.307a.625.625 0 1 1 .884.885l-3.75 3.749a.625.625 0 0 1-.884 0l-3.75-3.749a.626.626 0 0 1 0-.885Z"
          fill-rule="evenodd"
        />
      </symbol>
      <use
        href="#icon-chevron-down-12"
      />
    </svg>
  </button>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.e=[0,1,_.a={"EventAttributes:#text/0":_.b={},"ConditionalScope:#text/0":_.d={},"ConditionalRenderer:#text/0":"button",content:_._["src/tags/evo-filter-chip/examples/menu-button.marko_1_renderer"](_.c={}),variant:"menu",htmlInput:{},isMenu:!0,isAnchor:!1,mounted:!1,"ClosureScopes:selected":_.j=new Set},_.d,_.g={_:_.a,"ClosureSignalIndex:selected":0,"#ClosestBranchId":3},3,{htmlInput:{}}],_.b.click=_._["src/tags/evo-filter-chip/index.marko_0/onClick"](_.a),(_.f=new Set).add(_.g),(_.h=new Set).add(_.g),(_.i=new Set).add(_.g),(_.j).add(_.g),_.e),"src/tags/evo-icon/index.marko_0_htmlInput",8,"src/tags/evo-filter-chip/index.marko_0",2];M._.w()
  </script>
</DocumentFragment>
`;

exports[`evo-filter-chip > renders menu button expanded 1`] = `
<DocumentFragment>
  <button
    aria-expanded="true"
    aria-pressed="false"
    class="filter-chip"
    type="button"
  >
    <span
      class="filter-chip__text"
    >
      Filter
    </span>
    <svg
      aria-hidden="true"
      class="filter-chip__trailing icon icon--12"
    >
      <symbol
        id="GENERATED-0"
        viewBox="0 0 12 12"
      >
        <path
          clip-rule="evenodd"
          d="M1.808 4.188a.625.625 0 0 1 .884 0L6 7.495l3.308-3.307a.625.625 0 1 1 .884.885l-3.75 3.749a.625.625 0 0 1-.884 0l-3.75-3.749a.626.626 0 0 1 0-.885Z"
          fill-rule="evenodd"
        />
      </symbol>
      <use
        href="#icon-chevron-down-12"
      />
    </svg>
  </button>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.e=[0,1,_.a={"EventAttributes:#text/0":_.b={},"ConditionalScope:#text/0":_.d={},"ConditionalRenderer:#text/0":"button",content:_._["src/tags/evo-filter-chip/examples/menu-button.marko_1_renderer"](_.c={}),variant:"menu",htmlInput:{},isMenu:!0,isAnchor:!1,mounted:!1,expanded:!0,"ClosureScopes:selected":_.j=new Set},_.d,_.g={_:_.a,"ClosureSignalIndex:selected":0,"#ClosestBranchId":3},3,{htmlInput:{}}],_.b.click=_._["src/tags/evo-filter-chip/index.marko_0/onClick"](_.a),(_.f=new Set).add(_.g),(_.h=new Set).add(_.g),(_.i=new Set).add(_.g),(_.j).add(_.g),_.e),"src/tags/evo-icon/index.marko_0_htmlInput",8,"src/tags/evo-filter-chip/index.marko_0",2];M._.w()
  </script>
</DocumentFragment>
`;

exports[`evo-filter-chip > renders menu button expanded and selected 1`] = `
<DocumentFragment>
  <button
    aria-expanded="true"
    aria-pressed="true"
    class="filter-chip"
    type="button"
  >
    <span
      class="filter-chip__text"
    >
      Filter
    </span>
    <svg
      aria-hidden="true"
      class="filter-chip__trailing icon icon--12"
    >
      <symbol
        id="GENERATED-0"
        viewBox="0 0 12 12"
      >
        <path
          clip-rule="evenodd"
          d="M1.808 4.188a.625.625 0 0 1 .884 0L6 7.495l3.308-3.307a.625.625 0 1 1 .884.885l-3.75 3.749a.625.625 0 0 1-.884 0l-3.75-3.749a.626.626 0 0 1 0-.885Z"
          fill-rule="evenodd"
        />
      </symbol>
      <use
        href="#icon-chevron-down-12"
      />
    </svg>
  </button>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.e=[0,1,_.a={"EventAttributes:#text/0":_.b={},"ConditionalScope:#text/0":_.d={},"ConditionalRenderer:#text/0":"button",content:_._["src/tags/evo-filter-chip/examples/menu-button.marko_1_renderer"](_.c={}),variant:"menu",htmlInput:{},isMenu:!0,isAnchor:!1,mounted:!1,selected:!0,expanded:!0,"ClosureScopes:selected":_.j=new Set},_.d,_.g={_:_.a,"ClosureSignalIndex:selected":0,"#ClosestBranchId":3},3,{htmlInput:{}}],_.b.click=_._["src/tags/evo-filter-chip/index.marko_0/onClick"](_.a),(_.f=new Set).add(_.g),(_.h=new Set).add(_.g),(_.i=new Set).add(_.g),(_.j).add(_.g),_.e),"src/tags/evo-icon/index.marko_0_htmlInput",8,"src/tags/evo-filter-chip/index.marko_0",2];M._.w()
  </script>
</DocumentFragment>
`;

exports[`evo-filter-chip > renders menu button selected 1`] = `
<DocumentFragment>
  <button
    aria-expanded="false"
    aria-pressed="true"
    class="filter-chip"
    type="button"
  >
    <span
      class="filter-chip__text"
    >
      Filter
    </span>
    <svg
      aria-hidden="true"
      class="filter-chip__trailing icon icon--12"
    >
      <symbol
        id="GENERATED-0"
        viewBox="0 0 12 12"
      >
        <path
          clip-rule="evenodd"
          d="M1.808 4.188a.625.625 0 0 1 .884 0L6 7.495l3.308-3.307a.625.625 0 1 1 .884.885l-3.75 3.749a.625.625 0 0 1-.884 0l-3.75-3.749a.626.626 0 0 1 0-.885Z"
          fill-rule="evenodd"
        />
      </symbol>
      <use
        href="#icon-chevron-down-12"
      />
    </svg>
  </button>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.e=[0,1,_.a={"EventAttributes:#text/0":_.b={},"ConditionalScope:#text/0":_.d={},"ConditionalRenderer:#text/0":"button",content:_._["src/tags/evo-filter-chip/examples/menu-button.marko_1_renderer"](_.c={}),variant:"menu",htmlInput:{},isMenu:!0,isAnchor:!1,mounted:!1,selected:!0,"ClosureScopes:selected":_.j=new Set},_.d,_.g={_:_.a,"ClosureSignalIndex:selected":0,"#ClosestBranchId":3},3,{htmlInput:{}}],_.b.click=_._["src/tags/evo-filter-chip/index.marko_0/onClick"](_.a),(_.f=new Set).add(_.g),(_.h=new Set).add(_.g),(_.i=new Set).add(_.g),(_.j).add(_.g),_.e),"src/tags/evo-icon/index.marko_0_htmlInput",8,"src/tags/evo-filter-chip/index.marko_0",2];M._.w()
  </script>
</DocumentFragment>
`;

exports[`evo-filter-chip > renders static default 1`] = `
<DocumentFragment>
  <button
    aria-pressed="false"
    class="filter-chip"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="icon icon--16"
    >
      <symbol
        id="GENERATED-0"
        viewBox="0 0 16 16"
      >
        <path
          clip-rule="evenodd"
          d="M4.17 3A3.001 3.001 0 0 1 7 1c1.826 0 2.667 1.549 2.97 2.757l.394 1.575 4.073 2.222A3 3 0 0 1 16 10.187V11a3 3 0 0 1-3 3H2.5A2.5 2.5 0 0 1 0 11.5v-6A2.5 2.5 0 0 1 2.5 3h1.67ZM6 4a1 1 0 0 1 1-1c.629 0 .903.737 1.03 1.243l.263 1.05-1 1a1 1 0 0 0 1.414 1.414l.648-.648c.248.145.505.277.758.414l-.82.82a1 1 0 0 0 1.414 1.414l1.236-1.236 1.536.838a1 1 0 0 1 .521.878V11a1 1 0 0 1-1 1H7v-.5A2.5 2.5 0 0 0 4.5 9H2V8h1.5A2.5 2.5 0 0 0 6 5.5V4Zm-1 8v-.5a.5.5 0 0 0-.5-.5H2v.5a.5.5 0 0 0 .5.5H5ZM2.5 5H4v.5a.5.5 0 0 1-.5.5H2v-.5a.5.5 0 0 1 .5-.5Z"
          fill-rule="evenodd"
        />
      </symbol>
      <use
        href="#icon-sneaker-16"
      />
    </svg>
    <span
      class="filter-chip__text"
    >
      Filter
    </span>
  </button>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.e=[0,1,_.a={"EventAttributes:#text/0":_.b={},"ConditionalScope:#text/0":_.d={},"ConditionalRenderer:#text/0":"button",content:_._["src/tags/evo-filter-chip/examples/default.marko_1_renderer"](_.c={}),icon:{content:_._["src/tags/evo-filter-chip/examples/default.marko_2_renderer"](_.c),*[Symbol.iterator](){yield this}},variant:"default",htmlInput:{},isMenu:!1,isAnchor:!1,mounted:!1,"ClosureScopes:selected":_.j=new Set},_.d,_.g={_:_.a,"ClosureSignalIndex:selected":0,"#ClosestBranchId":3},3,{htmlInput:{}}],_.b.click=_._["src/tags/evo-filter-chip/index.marko_0/onClick"](_.a),(_.f=new Set).add(_.g),(_.h=new Set).add(_.g),(_.i=new Set).add(_.g),(_.j).add(_.g),_.e),"src/tags/evo-icon/index.marko_0_htmlInput",8,"src/tags/evo-filter-chip/index.marko_0",2];M._.w()
  </script>
</DocumentFragment>
`;

exports[`evo-filter-chip > renders static selected 1`] = `
<DocumentFragment>
  <button
    aria-pressed="true"
    class="filter-chip"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="icon icon--16"
    >
      <symbol
        id="GENERATED-0"
        viewBox="0 0 16 16"
      >
        <path
          clip-rule="evenodd"
          d="M4.17 3A3.001 3.001 0 0 1 7 1c1.826 0 2.667 1.549 2.97 2.757l.394 1.575 4.073 2.222A3 3 0 0 1 16 10.187V11a3 3 0 0 1-3 3H2.5A2.5 2.5 0 0 1 0 11.5v-6A2.5 2.5 0 0 1 2.5 3h1.67ZM6 4a1 1 0 0 1 1-1c.629 0 .903.737 1.03 1.243l.263 1.05-1 1a1 1 0 0 0 1.414 1.414l.648-.648c.248.145.505.277.758.414l-.82.82a1 1 0 0 0 1.414 1.414l1.236-1.236 1.536.838a1 1 0 0 1 .521.878V11a1 1 0 0 1-1 1H7v-.5A2.5 2.5 0 0 0 4.5 9H2V8h1.5A2.5 2.5 0 0 0 6 5.5V4Zm-1 8v-.5a.5.5 0 0 0-.5-.5H2v.5a.5.5 0 0 0 .5.5H5ZM2.5 5H4v.5a.5.5 0 0 1-.5.5H2v-.5a.5.5 0 0 1 .5-.5Z"
          fill-rule="evenodd"
        />
      </symbol>
      <use
        href="#icon-sneaker-16"
      />
    </svg>
    <span
      class="filter-chip__text"
    >
      Filter
    </span>
  </button>
  <script>
    ((runtimeId) =&gt; (
(self[runtimeId] = (
  renderId,
  prefix = runtimeId + renderId,
  prefixLen = prefix.length,
  lookup = {},
  visits = [],
  doc = document,
  walker = doc.createTreeWalker(
    doc,
    129 /* NodeFilter.SHOW_COMMENT | NodeFilter.SHOW_ELEMENT */,
  ),
) =&gt;
  doc = (self[runtimeId][renderId] = {
    i: prefix,
    d: doc,
    l: lookup,
    v: visits,
    x() {},
    w(node, op, id) {
      while ((node = walker.nextNode())) {
        doc.x(
          (op =
            (op = node.data) &&
            !op.indexOf(prefix) &&
            ((lookup[(id = op.slice(prefixLen + 1))] = node), op[prefixLen])),
          id,
          node,
        );

        if (op &gt; "#") {
          visits.push(node);
        }
      }
    },
  })),
self[runtimeId]
))("M")("_");M._.r=[_=&gt;(_.e=[0,1,_.a={"EventAttributes:#text/0":_.b={},"ConditionalScope:#text/0":_.d={},"ConditionalRenderer:#text/0":"button",content:_._["src/tags/evo-filter-chip/examples/default.marko_1_renderer"](_.c={}),icon:{content:_._["src/tags/evo-filter-chip/examples/default.marko_2_renderer"](_.c),*[Symbol.iterator](){yield this}},variant:"default",htmlInput:{},isMenu:!1,isAnchor:!1,mounted:!1,selected:!0,"ClosureScopes:selected":_.j=new Set},_.d,_.g={_:_.a,"ClosureSignalIndex:selected":0,"#ClosestBranchId":3},3,{htmlInput:{}}],_.b.click=_._["src/tags/evo-filter-chip/index.marko_0/onClick"](_.a),(_.f=new Set).add(_.g),(_.h=new Set).add(_.g),(_.i=new Set).add(_.g),(_.j).add(_.g),_.e),"src/tags/evo-icon/index.marko_0_htmlInput",8,"src/tags/evo-filter-chip/index.marko_0",2];M._.w()
  </script>
</DocumentFragment>
`;
